internal/containerupdate/exec_updater_test.go

```diff
@@ -0,0 +1,126 @@
+package containerupdate
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/windmilleng/tilt/internal/sliceutils"
+
+	"github.com/windmilleng/tilt/internal/k8s"
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/testutils"
+)
+
+var toDelete = []string{"/foo/delete_me", "/bar/me_too"}
+var (
+	cmdA = model.Cmd{Argv: []string{"a"}}
+	cmdB = model.Cmd{Argv: []string{"b", "bar", "baz"}}
+)
+var cmds = []model.Cmd{cmdA, cmdB}
+
+func TestUpdateContainerDoesntSupportRestart(t *testing.T) {
+	f := newExecFixture(t)
+
+	err := f.ecu.UpdateContainer(f.ctx, TestDeployInfo, newReader("boop"), toDelete, cmds, false)
+	if assert.NotNil(t, err, "expect Exec UpdateContainer to fail if !hotReload") {
+		assert.Contains(t, err.Error(), "ExecUpdater does not support `restart_container()` step")
+	}
+}
+
+func TestUpdateContainerDeletesFiles(t *testing.T) {
+	f := newExecFixture(t)
+
+	// No files to delete
+	err := f.ecu.UpdateContainer(f.ctx, TestDeployInfo, newReader("boop"), nil, cmds, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, call := range f.kCli.ExecCalls {
+		if sliceutils.StringSliceStartsWith(call.Cmd, []string{"rm", "-rf"}) {
```

Comment:
this test isn't super robust. it will stop testing the thing you're looking for if we change the flags to `rm`. 

a better way to test this would be to have a helper like f.getRmExecCalls() and use that helper in multiple places

or better yet, to do something like
f.assertExecCallsArg0([]string{"rm"})

Reply:
mmm good call. will change.